src/Compilers/VisualBasic/Portable/Symbols/Tuples/TupleTypeSymbol.vb

```diff
@@ -0,0 +1,987 @@
+
+        Public Overrides ReadOnly Property IsTupleType() As Boolean
+            Get
+                Return True
+            End Get
+        End Property
+
+        Public Overrides ReadOnly Property TupleUnderlyingType() As NamedTypeSymbol
+            Get
+                Return Me._underlyingType
+            End Get
+        End Property
+
+        Public Overrides ReadOnly Property TupleElementTypes() As ImmutableArray(Of TypeSymbol)
+            Get
+                Return Me._elementTypes
+            End Get
+        End Property
+
+        Public Overrides ReadOnly Property TupleElementNames() As ImmutableArray(Of String)
+            Get
+                Return Me._elementNames
+            End Get
+        End Property
+
+        Public Overrides ReadOnly Property IsReferenceType() As Boolean
+            Get
+                Return Not Me._underlyingType.IsErrorType() AndAlso Me._underlyingType.IsReferenceType
+            End Get
+        End Property
+
+        Public Overrides ReadOnly Property IsValueType() As Boolean
+            Get
+                Return Me._underlyingType.IsValueType
+            End Get
+        End Property
+
+        Public Overrides ReadOnly Property IsImplicitlyDeclared() As Boolean
+            Get
+                Return False
+            End Get
+        End Property
+
+        Public ReadOnly Property TupleElementFields() As ImmutableArray(Of FieldSymbol)
+            Get
+                Dim isDefault As Boolean = Me._lazyFields.IsDefault
+                If isDefault Then
+                    ImmutableInterlocked.InterlockedInitialize(Of FieldSymbol)(Me._lazyFields, Me.CollectTupleElementFields())
+                End If
+                Return Me._lazyFields
+            End Get
+        End Property
+
+        Friend ReadOnly Property UnderlyingDefinitionToMemberMap() As SmallDictionary(Of Symbol, Symbol)
+            Get
+                If Me._lazyUnderlyingDefinitionToMemberMap Is Nothing Then
+                    Me._lazyUnderlyingDefinitionToMemberMap = Me.ComputeDefinitionToMemberMap()
+                End If
+                Return Me._lazyUnderlyingDefinitionToMemberMap
+            End Get
+        End Property
+
+        Public Overrides ReadOnly Property EnumUnderlyingType() As NamedTypeSymbol
+            Get
+                Return Me._underlyingType.EnumUnderlyingType
+            End Get
+        End Property
+
+        Public Overrides ReadOnly Property Kind() As SymbolKind
+            Get
+                Return SymbolKind.NamedType
+            End Get
+        End Property
+
+        Public Overrides ReadOnly Property TypeKind() As TypeKind
+            Get
+                Return If((Me._underlyingType.TypeKind = TypeKind.[Class]), TypeKind.[Class], TypeKind.Struct)
```

Comment:
done

---
In reply to: [71552461](https://github.com/dotnet/roslyn/pull/12596#discussion_r71552461) [](ancestors = 71552461)